### ISSUE TITLE
push.clearAllNotifications does not work in Ionic v2

### ISSUE THREAD

**Issue by uj** [phonegap/phonegap-plugin-push#1530]
Ionic v2 RC4
phonegap-plugin-push v1.8.4

ionic plugin add phonegap-plugin-push@1.8.4 --variable SENDER_ID=xxxxxxxxxx

import {Push} from "ionic-native"

let push=Push.init(....) <--- THIS WORKS

push.on(...) <--- THIS WORKS
push.finish(...) <--- THIS WORKS

I can successfully receive push notifications.  The problem is that I cannot clear the notifications from the app, as though "clearAllNotifications" is not an available option.

push.clearAllNotifications(...) <-- ERROR HERE:   Property 'clearAllNotifications' does not exist on type 'PushNotification'.

**Comment by fredgalvao** [comment]
That's mostly an issue on ionic-native, and partly an issue on our Typescript typings (which is outdated).

I've scanned through ionic-native source code and I **guess** they are doing a typescript scan on the interface to pick the methods to mirror, and then the lack of the type definition for the `clearAllNotifications` method [on DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/phonegap-plugin-push/index.d.ts), made ionic use the [wrong coppied typings](https://github.com/driftyco/ionic-native/blob/0d6997cbdd717471e190b8282f6096922d197692/src/plugins/push.ts), and then you get your error.

Basically, it's a typescript/ionic mistake, not an actual error with the plugin.

**Comment by macdonst** [comment]
@fredgalvao anything we can do to fix this in our TypeScript binding or should we just close it?

**Comment by fredgalvao** [comment]
I'll open an issue to mirror the current state of the api and settings/options to DefinitelyTyped, and also review some stuff for ts@2.1, but I doubt this will automagically fix things on ionic's end. We still need the guys over there to either [wait for my update, then copy the updated file over to ionic-native] or [just update ionic-native directly].

**Comment by fredgalvao** [comment]
I consider this `closed` as per the fact that fixing issue #1536 won't fix this, and thus the main and final issue is still with ionic-native. In any way, please follow that issue for further details.

**Comment by jakute** [comment]
hello there, im new here. 
I have the same issue, im using ionic 3, phonegap-plugin-push 1.1.2.
and i can't clear notifications when the user clicks on the icon app. i tried to use clearallnotificatiosn() but shows me the same error : Property 'clearAllNotifications' does not exist on type 'PushNotification'. I saw the file and the code is correct.

thanks in advance

**Comment by fredgalvao** [comment]
@jakute  If the code "is correct", then the issue is somewhere else. If you have more info than that, please open a new issue and provide some detailed context as well as steps to reproduce that.
